ui/src/variables/actions/index.ts

```diff
@@ -179,18 +204,24 @@ export const createVariable = (
 ) => async (dispatch: Dispatch<Action>, getState: GetState) => {
   try {
     const org = getOrg(getState())
-    const createdVariable = await client.variables.create({
-      ...variable,
-      orgID: org.id,
+    const resp = await apiPostVariable({
+      data: {
+        ...variable,
+        orgID: org.id,
+      },
     })
 
-    dispatch(
-      setVariable(createdVariable.id, RemoteDataState.Done, createdVariable)
-    )
+    if (resp.status !== 201) {
+      throw new Error(resp.data.message)
+    }
+
+    const createdVar = addVariableDefaults(resp.data)
+
+    dispatch(setVariable(createdVar.id, RemoteDataState.Done, createdVar))
     dispatch(notify(createVariableSuccess(variable.name)))
   } catch (e) {
     console.error(e)
-    dispatch(notify(createVariableFailed(e.response.data.message)))
+    dispatch(notify(createVariableFailed(e.message)))
```

Comment:
make sure you have tested this please

Reply:
This actually returned an error when it was set as `e.response.data.message` since the error we're throwing is `e.message`